pkgs/by-name/cr/crow/package.nix

```
@@ -0,0 +1,42 @@
+{
+  lib,
+  stdenv,
+  runCommand,
+  fetchFromGitHub,
+  cmake,
+  asio,
+  python3,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "crow";
+  version = "1.2";
+
+  src = fetchFromGitHub {
+    owner = "crowcpp";
+    repo = "crow";
+    rev = "v${version}";
+    sha256 = "sha256-fokj+KiS6frPVOoOvETxW3ue95kCcYhdhOlN3efzBd4=";
+  };
+
+  propagatedBuildInputs = [ asio ];
+  nativeBuildInputs = [
+    asio
+    cmake
+    python3
+  ];
+
+  cmakeFlags = [
+    "-DCROW_BUILD_EXAMPLES=OFF"
+  ];
+
+  doCheck = true;
+
+  meta = with lib; {
+    description = "A Fast and Easy to use microframework for the web";
+    homepage = "https://crowcpp.org/";
+    maintainers = with maintainers; [ l33tname ];
+    platforms = platforms.all;
+    license = licenses.bsd3;
```

```suggestion
  meta = {
    description = "A Fast and Easy to use microframework for the web";
    homepage = "https://crowcpp.org/";
    maintainers = with lib.maintainers; [ l33tname ];
    platforms = lib.platforms.all;
    license = lib.licenses.bsd3;
```

It's better to avoid `with lib;` whenever possible.

---

I lifted that from [pkgs/development/libraries/argparse/default.nix](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/argparse/default.nix) is there a link or something explaining why i should not use `with` there?

---

https://github.com/NixOS/nixpkgs/issues/208242

It's generally fine to have it in a small scope like this but still.

Let's say you are trying to use `version` variable from the package attrset, if `rec` is not used and the expression is in the scope of `with lib;`, the expression will still eval, instead of erroring out, having got the `version` from `lib`.


As for `rec` itself, it is [discouraged to use](https://github.com/NixOS/nixpkgs/issues/315337#issuecomment-2138814370), `finalAttrs` should be used whenever possible.

---

updated to use finalAttrs :+1: